lib/puppet/type/sqlserver_instance.rb

```diff
@@ -158,7 +158,7 @@ def strong_password?(key)
     failures << 'must contain lowercase letters' unless password =~ %r{[a-z]}
     failures << 'must contain uppercase letters' unless password =~ %r{[A-Z]}
     failures << 'must contain numbers' unless password =~ %r{\d}
-    failures << 'must contain a special character' unless password =~ %r{[^A-Za-z0-9]}
+    failures << 'must contain a special character' unless password =~ %r{}
```

Comment:
What does this check actually do now?

Reply:
@michaeltlombardi As far as I can see, nothing. This matches anything, from my testing. I think this pattern (`%r{[^A-Za-z0-9]}`) will fit the bill for us, however, @DavidS was advising me to separate bug fixes / feature work to a separate PR / ticket: [MODULES-10064](https://tickets.puppetlabs.com/browse/MODULES-10064).

There's the question round performing any password complexity checks in our domain, and if we do, should we use a library to do so, etc... There's details on the ticket.

Reply:
@cmccrisken-puppet separating rubocop fixes and bugfixes did not mean to *delete* the validation here ...

Reply:
Granted, @DavidS, but we still have the overarching question of whether we want to do it at all? However, I can add this in, if people feel it's A) useful and B) valid.

Reply:
oh - I'm confused by the diff here the original code has the empty regex (see current master: https://github.com/puppetlabs/puppetlabs-sqlserver/blob/a6c21653a1cfaaa06fd4574db34784b7f3e8c6b8/lib/puppet/type/sqlserver_instance.rb#L193) and therefore this also shouldn't have. As you said correctly the other decision&(potential)patch is captured in MODULES-10064

Reply:
Ah yes - it's not exactly as before due to the Rubocop fixes. Make the useless check Rubocop compliant 😆

Reply:
present consistency > future functionality